waspc/src/Generator/PackageJsonGenerator.hs

```diff
@@ -58,8 +59,16 @@ resolveNpmDeps waspDeps userDeps =
     isDepWithNameInWaspDeps :: String -> Bool
     isDepWithNameInWaspDeps name = any ((name ==) . ND._name) waspDeps
 
-toPackageJsonDependenciesString :: [ND.NpmDependency] -> String
-toPackageJsonDependenciesString deps =
-  "\"dependencies\": {"
+dependencyToString :: [ND.NpmDependency] -> String -> String
+dependencyToString deps depType =
```

Comment:
Great! I would suggest renaming this to `npmDepsToPackageJsonEntryWithKey`, where entry indicates it is key + value, and then I would actually also rename `toPackageJsonDependenciesString` to `npmDepsToPackageJsonEntry` and `toPackageJsonDevDependenciesString` to `npmDevDepsToPackageJsonEntry` and those would be real nice names all together. And then you can rename `depType` argument to `key`.

Btw alternative approach might have been to create `npmDepsToPackageJsonMap` which would return `{ ... }` without the key, and then in `toPackageJsonDependenciesString` you would do `= "\"dependencies\": " ++ (npmDepsToPackageJsonMap deps)`, which this does leave a little bit of duplication on the table but it also makes that helper function somewhat easier to reason about. Your approach is totally fine though, let's go with it.

Reply:
Done!